server/mediaserver.go

```diff
@@ -188,35 +188,52 @@ func gotRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.
 			glog.Errorf("Error creating playlist: %v", err)
 			return ErrRTMPPublish
 		}
-		hlsStrm, err := s.LivepeerNode.VideoDB.AddNewHLSStream(hlsStrmID)
+
+		//Get a broadcaster from the network
+		broadcaster, err := s.LivepeerNode.VideoNetwork.GetBroadcaster(string(hlsStrmID))
 		if err != nil {
 			glog.Errorf("Error creating HLS stream for segmentation: %v", err)
 			return ErrRTMPPublish
 		}
 		LastHLSStreamID = hlsStrmID
 
-		//Segment the stream (this populates the hls stream)
-		go func() {
+		//Segment the stream, insert the segments into the broadcaster
+		go func(broadcaster stream.Broadcaster, rtmpStrm stream.RTMPVideoStream) {
+			hlsStrm := stream.NewBasicHLSVideoStream(string(hlsStrmID), stream.DefaultHLSStreamWin)
+			hlsStrm.SetSubscriber(func(seg *stream.HLSSegment, eof bool) {
+				if eof {
+					broadcaster.Finish()
+					return
+				}
+
+				segHash := (&ethTypes.Segment{StreamID: hlsStrm.GetStreamID(), SegmentSequenceNumber: big.NewInt(int64(seg.SeqNo)), DataHash: crypto.Keccak256Hash(seg.Data)}).Hash()
```

Comment:
Could this callback logic just be a call to `s.LivepeerNode.BroadcastHLSSegToNetwork(...)`?